doc/source/whatsnew/v1.0.0.rst

```diff
@@ -78,6 +78,7 @@ Performance improvements
 Bug fixes
 ~~~~~~~~~
 
+- Previously, when :class:`pandas.core.groupby.GroupBy.transform` was passed the name of a transformation (e.g. `rank`, `ffill`, etc) it broadcast the results resulting in erroneous values. Only reduction results and not the results of transformations. The return value in these cases is the same as calling the named function directly (e.g. `g.rank()`) (:issue:`14274`) (:issue:`19354`) (:issue:`22509`).
```

Comment:
this would need a full sub-section with example of previous and new, as this actually is a non-trivial api breaking change. (so put in that section)

Reply:
Breaking change or bug? I don't understand how broadcasting the result of `rank` could have ever been intended behavior. I can understand `g.transform('rank')` being an alias for `g.rank()`, and I can understand `g.transform('rank')` raising and telling the user to use `g.rank()` directly (as as suggested), but I don't see how any other behavior could be intended. 

Please correct me if I'm wrong.

Reply:
Or are you agreeing its bug, and just saying this is a big deal and we should feature this fix more prominently?

Reply:
It's both but sends a clearer message if put in the API Breaking Changes section

Reply:
Sorry, I misunderstood. Raising on transformations is definitely a breaking change. Will do.